library/nmcli.py

```diff
@@ -0,0 +1,735 @@
+        elif self.type=='vlan':
+            cmd=self.modify_connection_vlan()
+        return self.execute_command(cmd)
+
+
+def main():
+    # Parsing argument file
+    module=AnsibleModule(
+        argument_spec=dict(
+            enabled=dict(required=False, default=None, choices=['yes', 'no'], type='str'),
+            name=dict(required=False, default=None, choices=['add', 'mod', 'show', 'up', 'down', 'del'], type='str'),
+            state=dict(default=None, choices=['present', 'absent', 'modify'], type='str'),
+            cname=dict(required=False, type='str'),
+            master=dict(required=False, default=None, type='str'),
+            autoconnect=dict(required=False, default=None, choices=['yes', 'no'], type='str'),
+            ifname=dict(required=False, default=None, type='str'),
+            type=dict(required=False, default=None, choices=['ethernet', 'team', 'team-slave', 'bond', 'bond-slave', 'bridge', 'vlan'], type='str'),
+            ip4=dict(required=False, default=None, type='str'),
+            gw4=dict(required=False, default=None, type='str'),
+            dns4=dict(required=False, default=None, type='str'),
+            ip6=dict(required=False, default=None, type='str'),
+            gw6=dict(required=False, default=None, type='str'),
+            dns6=dict(required=False, default=None, type='str'),
+            # general usage
+            mtu=dict(required=False, default=None, type='str'),
+            mac=dict(required=False, default=None, type='str'),
+            # bridge specific vars
+            stp=dict(required=False, default='yes', choices=['yes', 'no'], type='str'),
+            priority=dict(required=False, default="128", type='str'),
+            slavepriority=dict(required=False, default="32", type='str'),
+            forwarddelay=dict(required=False, default="15", type='str'),
+            hellotime=dict(required=False, default="2", type='str'),
+            maxage=dict(required=False, default="20", type='str'),
+            ageingtime=dict(required=False, default="300", type='str'),
+            # vlan specific vars
+            vlanid=dict(required=False, default=None, type='str'),
+            # following options are specific to ifnamedel
+        ),
+        supports_check_mode=True
+    )
+
+    # if module.params['state'] is None and module.params['enabled'] is None:
+    #     module.fail_json(msg="'Neither state' nor 'enabled' is set")
+
+    nmcli=Nmcli(module)
+
+    if nmcli.syslogging:
+        syslog.openlog('ansible-%s' % os.path.basename(__file__))
+        syslog.syslog(syslog.LOG_NOTICE, 'Nmcli instantiated - platform %s' % nmcli.platform)
+        if nmcli.distribution:
+            syslog.syslog(syslog.LOG_NOTICE, 'Nuser instantiated - distribution %s' % nmcli.distribution)
+
+    rc=None
+    out=''
+    err=''
+    result={}
+    result['cname']=nmcli.cname
+    result['state']=nmcli.state
+
+    if nmcli.state=='absent':
+        # if nmcli.connection_exists():
```

Comment:
I was getting some weird happenings the other day. connection_exists() does now work. Will put that test back in.